src/miner.cpp

```diff
@@ -251,9 +251,10 @@ CBlockTemplate* Mining::CreateNewBlock(const CChainParams& chainparams) const
             if (!IsFinalTx(tx, nHeight, nLockTimeCutoff))
                 continue;
 
+            const uint64_t maxSigOps = Policy::blockSigOpAcceptLimit(nBlockSize + nTxSize);
```

Comment:
This is absolutely huge and may win being rearchitectured. I've been scrolling several screens back and still can't figure out the whole picture. My understanding is that nBlockSize accumulate the size of the block and nTxSize the size of the tx currently being considered for inclusion. Please correct me if that's wrong.

First, there is a problem because nBlockSize reserve space for the coinbase, and the sigops limit may end up being different in case the coinbase being used by the miner is smaller than what was reserved. I would argue that the size reserved for the coinbase shouldn't be accounted for when computing the sigops limit to be conservative.